### ISSUE TITLE
Docker for Mac Kubernetes startup errors after EDGE upgrade

### ISSUE THREAD

**Issue by wenlock** [docker/for-mac#2797]
Please, check https://docs.docker.com/docker-for-mac/troubleshoot/.
Issues without logs and details cannot be debugged, and will be closed.

  - [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have submitted Diagnostics
  - [x] I have included the Diagnostics ID in this Issue

### Expected behavior
When enabling kubernetes option on the latest release, Kubernetes should start within an hour.

### Actual behavior

I've tried full uninstalling and re-installing docker, removing all signs of docker on my mac before a new install. However, after doing a fresh install of d4mk on 18.04.0-ce-rc2, kubernetes is not starting.

Looking at some of the container logs, I found the following had this error message:

```
I0410 20:04:25.853169       1 logs.go:41] http: TLS handshake error from 10.1.1.3:46808: remote error: tls: bad certificate
```

Here is a slightly longer log output
```
docker logs k8s_kube-apiserver_kube-apiserver-docker-for-desktop_kube-system_538c47cf5fd641e5098113fc61fd3f6b_0
I0410 20:04:23.120913       1 server.go:121] Version: v1.9.6
I0410 20:04:23.217800       1 feature_gate.go:190] feature gates: map[Initializers:true]
I0410 20:04:23.217836       1 initialization.go:90] enabled Initializers feature as part of admission plugin setup
I0410 20:04:23.222991       1 master.go:225] Using reconciler: master-count
W0410 20:04:23.271751       1 genericapiserver.go:342] Skipping API batch/v2alpha1 because it has no resources.
W0410 20:04:23.284003       1 genericapiserver.go:342] Skipping API rbac.authorization.k8s.io/v1alpha1 because it has no resources.
W0410 20:04:23.284859       1 genericapiserver.go:342] Skipping API storage.k8s.io/v1alpha1 because it has no resources.
W0410 20:04:23.325246       1 genericapiserver.go:342] Skipping API admissionregistration.k8s.io/v1alpha1 because it has no resources.
[restful] 2018/04/10 20:04:23 log.go:33: [restful/swagger] listing is available at https://10.1.1.3:6443/swaggerapi
[restful] 2018/04/10 20:04:23 log.go:33: [restful/swagger] https://10.1.1.3:6443/swaggerui/ is mapped to folder /swagger-ui/
[restful] 2018/04/10 20:04:23 log.go:33: [restful/swagger] listing is available at https://10.1.1.3:6443/swaggerapi
[restful] 2018/04/10 20:04:23 log.go:33: [restful/swagger] https://10.1.1.3:6443/swaggerui/ is mapped to folder /swagger-ui/
I0410 20:04:25.849349       1 serve.go:89] Serving securely on [::]:6443
I0410 20:04:25.849485       1 controller.go:84] Starting OpenAPI AggregationController
I0410 20:04:25.849640       1 apiservice_controller.go:112] Starting APIServiceRegistrationController
I0410 20:04:25.849677       1 cache.go:32] Waiting for caches to sync for APIServiceRegistrationController controller
I0410 20:04:25.853169       1 logs.go:41] http: TLS handshake error from 10.1.1.3:46808: remote error: tls: bad certificate
I0410 20:04:25.853239       1 crd_finalizer.go:242] Starting CRDFinalizer
I0410 20:04:25.853831       1 available_controller.go:262] Starting AvailableConditionController
I0410 20:04:25.853853       1 cache.go:32] Waiting for caches to sync for AvailableConditionController controller
I0410 20:04:25.853867       1 crdregistration_controller.go:110] Starting crd-autoregister controller
I0410 20:04:25.853871       1 controller_utils.go:1019] Waiting for caches to sync for crd-autoregister controller
I0410 20:04:25.865981       1 logs.go:41] http: TLS handshake error from 10.1.1.3:46794: remote error: tls: bad certificate
I0410 20:04:25.872002       1 logs.go:41] http: TLS handshake error from 10.1.1.3:46798: remote error: tls: bad certificate
I0410 20:04:25.878276       1 customresource_discovery_controller.go:152] Starting DiscoveryController
I0410 20:04:25.878318       1 naming_controller.go:274] Starting NamingConditionController
I0410 20:04:25.879495       1 logs.go:41] http: TLS handshake error from 10.1.1.3:46788: remote error: tls: bad certificate
I0410 20:04:25.879572       1 logs.go:41] http: TLS handshake error from 10.1.1.3:46786: remote error: tls: bad certificate
I0410 20:04:25.879606       1 logs.go:41] http: TLS handshake error from 10.1.1.3:46782: remote error: tls: bad certificate
I0410 20:04:25.880015       1 logs.go:41] http: TLS handshake error from 10.1.1.3:46796: remote error: tls: bad certificate
I0410 20:04:25.880054       1 logs.go:41] http: TLS handshake error from 10.1.1.3:46792: remote error: tls: bad certificate
I0410 20:04:25.880076       1 logs.go:41] http: TLS handshake error from 10.1.1.3:46772: remote error: tls: bad certificate
I0410 20:04:25.881042       1 logs.go:41] http: TLS handshake error from 10.1.1.3:46778: remote error: tls: bad certificate
I0410 20:04:25.881115       1 logs.go:41] http: TLS handshake error from 10.1.1.3:46804: remote error: tls: bad certificate
I0410 20:04:25.881558       1 logs.go:41] http: TLS handshake error from 10.1.1.3:46784: remote error: tls: bad certificate
I0410 20:04:25.881828       1 logs.go:41] http: TLS handshake error from 10.1.1.3:46776: remote error: tls: bad certificate
I0410 20:04:25.886270       1 logs.go:41] http: TLS handshake error from 10.1.1.3:46790: remote error: tls: bad certificate
I0410 20:04:25.886645       1 logs.go:41] http: TLS handshake error from 10.1.1.3:46780: remote error: tls: bad certificate
```
Did not have this issue on 18.03.0

### Information

```
Docker for Mac: version: 18.04.0-ce-rc2-mac61 (7e2bb492d35dc35ec0037d8024b7e7be57fc72a9)
macOS: version 10.13.4 (build: 17E199)
logs: /tmp/93BC38C4-ABB6-4C75-AA4F-1D0E4224CE02/20180410-135119.tar.gz
[OK]     vpnkit
[OK]     virtualization hypervisor
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     kubernetes
[OK]     files
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     osxfs
[OK]     moby-console
[OK]     logs
[OK]     docker-cli
[OK]     disk

```

### Steps to reproduce the behavior

  1. Install latest edge version
  2. Enable kubernetes
  3. Stays stuck on starting

**Comment by wenlock** [comment]
One point I didn't mention. I've been having to change Docker subnet to 10.1.1.0 to deal with my VPN, maybe this is causing some conflicts in kube that didn't exist in the last patch.

**Comment by jan-g** [comment]
Seeing similar stuff arise all over the place: helm 2.9.x doesn't like this certificate. Also having problems with the coreos prometheus operator: it reports the same error when attempting to query for services to monitor.

**Comment by aaronsssya** [comment]
I got same issue. After I changed Docker subnet to 172.16.29.0/16. The Kubernetes is hang at "Kubernetes is starting..." phase. From the k8s_kube-controller container log, I discover the following error

`E0824 09:30:31.519028       1 leaderelection.go:224] error retrieving resource lock kube-system/kube-controller-manager: Get https://172.16.29.3:6443/api/v1/namespaces/kube-system/endpoints/kube-controller-manager: x509: certificate is valid for 10.96.0.1, 0.0.0.0, 192.168.65.3, 127.0.0.1, not 172.16.29.3`

Is there any way I can change the kube init command line? I believe default cmdlet or yaml should be changed for new subnet.
